include/kernel.h

```diff
@@ -4607,6 +4607,8 @@ extern int k_mem_slab_init(struct k_mem_slab *slab, void *buffer,
  *
  * This routine allocates a memory block from a memory slab.
  *
+ * @note Can be called by ISRs, but @a timeout must be set to K_NO_WAIT.
```

Comment:
how is this change related to the Kconfig changes in the title of this PR?

Reply:
They are not related, for some reason my last PR changes got submitted along with this one.  Pushed the changes.